### ISSUE TITLE
DateTimeImmutable::createFromFormat with ! returns DateTime

### ISSUE THREAD

**Issue by wizbit** [hnw/php-timecop#30]
When creating a datetimeimmutable object with ! in the format (to zero the time). I get a DateTime object returned.

Test: echo get_class(\DateTimeImmutable::createFromFormat('!Y-m-d', '2017-10-03'));
expected: DateTimeImmutable
got: DateTime

**Comment by Furgas** [comment]
Yeah, I thought I found some weird bug in PHP. Fortunately I rather quickly found out that it was timecop's fault. I guess we're not going to see it fixed soon :(
